### PR TITLE
Issue #278: YCSB does not throw error if there is no table created

### PR DESCRIPTION
**Pull request by gkamat**
[hbase] Verify that the table exists during DB binding initialization
and bail out if not found.  If this is not checked here, the workload
will continue running and generating errors until it finishes.

### CODE REVIEW

File: hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java [hunk 1]
```diff
@@ -133,6 +135,20 @@ public void init() throws DBException
...
         _columnFamilyBytes = Bytes.toBytes(_columnFamily);
+
+      // Terminate right now if table does not exist, since the client
+      // will not propagate this error upstream once the workload
+      // starts.
+      String table = com.yahoo.ycsb.workloads.CoreWorkload.table;
+      try
+	  {
+	      HTable ht = new HTable(config, table);
+	      HTableDescriptor dsc = ht.getTableDescriptor();
```
Comment (cmccoy) [inline hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java]: We might want to use [`Connection.getTable(...).getTableDescriptor()`](https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/client/Table.html#getTableDescriptor()) for the 1.0 client.

Comment (gkamat) [inline hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java]: Updated the change for the 1.0 client.

File: hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient.java [hunk 2]
```diff
@@ -99,6 +100,19 @@ public void init() throws DBException
...
+      // starts.
+      String table = com.yahoo.ycsb.workloads.CoreWorkload.table;
+      try
+	  {
+	      HTable ht = new HTable(config, table);
+	      HTableDescriptor dsc = ht.getTableDescriptor();
+	  }
+      catch (IOException e)
+	  {
+	      throw new DBException("Error accessing HBase table: " + table);
```
Comment (busbey) [inline hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient.java]: this should pass the source IOException in as the cause.

File: hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java [hunk 3]
```diff
@@ -133,6 +135,20 @@ public void init() throws DBException
...
+      // starts.
+      String table = com.yahoo.ycsb.workloads.CoreWorkload.table;
+      try
+	  {
+	      final TableName tableName = TableName.valueOf(table);
+	      HTableDescriptor dsc = _connection.getTable(tableName).getTableDescriptor();
+	  }
+      catch (IOException e)
+	  {
+	      throw new DBException("Error accessing HBase table: " + table);
```
Comment (busbey) [inline hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java]: this should pass the source IOException in as the cause.

**Comment by busbey** [comment]
+1

**Pull request merged**